### DISCUSSION TITLE
[Help]: Does thumbnails support rtl direction?

### DISCUSSION THREAD

**Discussion by hadikman** [davidjerleke/embla-carousel#782] [Help] [answered]
### Summary

Hi
At first I must say thank you for this lightweight and cool carousel library.

I'm doing a project with Nextjs 14 that supports right to left direction.
Everything is fine with the main slider and it's compatible well with **RTL**, but about thumbnails:

- Dragging the thumbnails to the end doesn't work and you can't see the last items are out of the viewport.
- Clicking on the slider and showing the current slider is placed out of the viewport doesn't work either.

I did this:
> Note: When using rtl, the content direction also has to be set to RTL, either by using the [HTML dir attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/dir) or the [CSS direction](https://developer.mozilla.org/en-US/docs/Web/CSS/direction) property.

I also added **direction: rtl** to the thumbnails container but it didn't help.

Thank you in advance

### If applicable, which variants of Embla Carousel are relevant to this question?

- [ ] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment]
Hi @hadikman,

Thank you for your question. Yes it does. Did you set both the main and thumbnails carousel [direction](https://www.embla-carousel.com/api/options/#direction) option to `rtl`?

**Reply by hadikman** [reply]
Yes, I did and also set **direction: rtl** in options

**Reply by davidjerleke** [reply]
@hadikman please create a simple CodeSandbox with your setup so I can see what’s going on.

Best,
David

**Comment by hadikman** [comment, marked as answer]
I realized that I missed to enable **direction: 'rtl'** in thumbnails options, so must enable in both useEmblaCarousel hooks:

`const [emblaMainRef, emblaMainApi] = useEmblaCarousel({
    direction: 'rtl',
    ...mainOptions,
  })`
`const [emblaThumbsRef, emblaThumbsApi] = useEmblaCarousel({
    direction: 'rtl',
    ...thumbnailsOptions,
  })`

**Reply by davidjerleke** [reply]
@hadikman yes that's what I asked about in my [previous comment](https://github.com/davidjerleke/embla-carousel/discussions/782#discussioncomment-8650146) 🙂:

> Thank you for your question. Yes it does. Did you set both the main and thumbnails carousel [direction](https://www.embla-carousel.com/api/options/#direction) option to `rtl` too?

Glad to hear that you figured it out.

**Reply by hadikman** [reply]
Yes, you right 😁. I read fast and I didn't notice.

**Comment by sarussss** [comment]
Hi @davidjerleke  @hadikman
When using RTL, my best choice is to use the [global options](https://www.embla-carousel.com/api/options/#global-options) because most likely you will want to enable RTL for all sliders on the website, you will not have to bother adding that option to the entire slider anymore, clean code. And when you need to disable RTL, you only need to change it once.

**Reply by hadikman** [reply]
Thank you very much, I did that 🙏🏼
